### PR TITLE
Extract transforms and utils

### PR DESCRIPTION
**Pull request by simonihmig**
Preparation work for #3: separated existing (classic component and template) transforms and utils into separate modules, so we can later add a native class transform easier.

### CODE REVIEW

**Comment by rajasegar** [comment]
Great job @simonihmig, I thought of doing the same. You got there first, the project looks more organized now by the way.

File: lib/transform.js [hunk 1]
```diff
@@ -87,235 +42,34 @@ function transform(source, template, options = {}) {
...
-    let [value, attr] = binding.split(':');
-    attrBindings.set(attr || value, value);
+  switch (type) {
+    case 'classic':
+      result = transformClassicComponent(root, options);
+      break;
+    default:
+      throw new Error(
+        `Unsupported component type. Only classic components (\`Component.extend({ ... }\`) are supported currently.`
+      );
```
Comment (Turbo87) [inline lib/transform.js]: why `Error` instead of `SilentError`?

File: lib/transform.js [hunk 2]
```diff
@@ -87,235 +42,34 @@ function transform(source, template, options = {}) {
...
+  let root = j(source);
+  let type = checkComponentType(root);
+  let result;
 
-function findAttributeBindings(properties) {
-  let attrBindings = new Map();
-  for (let binding of findStringArrayProperties(properties, 'attributeBindings')) {
-    let [value, attr] = binding.split(':');
-    attrBindings.set(attr || value, value);
+  switch (type) {
```
Comment (Turbo87) [inline lib/transform.js]: can you change this to be an if-condition? I'm slightly allergic to `switch`... 😅

File: lib/transform/classic.js [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+        isProperty(path, 'elementId') ||
+        isProperty(path, 'attributeBindings') ||
+        isProperty(path, 'classNames') ||
+        isProperty(path, 'classNameBindings')
+    )
+    .remove();
+
+  let newSource = root.toSource();
+
+  // wrap existing template with root element
```
Comment (Turbo87) [inline lib/transform/classic.js]: I'm wondering if everything below should rather be in the `transformTemplate()` function instead 🤔

File: lib/transform.js (hunk 1 above)
Comment (simonihmig) [reply lib/transform.js]: Because... working too late in the evening? 🤷‍♂️Fixed :wink:

File: lib/transform.js (hunk 2 above)
Comment (simonihmig) [reply lib/transform.js]: Sure, done.

File: lib/transform/classic.js (hunk 3 above)
Comment (simonihmig) [reply lib/transform/classic.js]: Makes sense! Keeps the template AST builder stuff isolated in that transform.

**Comment by simonihmig** [comment]
@Turbo87 Thanks for your feedback, updated the PR accordingly!

File: lib/transform/template.js [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+const PLACEHOLDER = '@@@PLACEHOLDER@@@';
+
+module.exports = function transformTemplate(
+  template,
+  tagName,
+  elementId,
+  classNames,
+  classNameBindings,
+  attributeBindings,
```
Comment (Turbo87) [inline lib/transform/template.js]: can we change these to be "named" parameters instead?

see https://github.com/ryanmcdermott/clean-code-javascript#function-arguments-2-or-fewer-ideally 😉

Comment (simonihmig) [reply lib/transform/template.js]: Good catch, done. I remember I wanted to clean this up, after making all tests to pass, but then apparently forgot...

File: lib/transform.js [hunk 5]
```diff
@@ -87,235 +42,47 @@ function transform(source, template, options = {}) {
...
+      classNameBindings,
+      attributeBindings,
+    } = result;
+    let newTemplate = transformTemplate({
+      template,
+      tagName,
+      elementId,
+      classNames,
+      classNameBindings,
+      attributeBindings,
```
Comment (simonihmig) [inline lib/transform.js]: Note: I cannot use object spread here because that's unsupported in node 8. But I guess we can drop support in a separate PR?

Comment (Turbo87) [reply lib/transform.js]: I've taken the liberty of simplifying it a little more, by moving the `template` and `option` arguments out of the named arguments block again. That way, we can directly pass in `result` without the need for object spread.

Comment (Turbo87) [reply lib/transform.js]: btw I had dropped Node 8 support in https://github.com/ember-codemods/tagless-ember-components-codemod/pull/48 😉

**Comment by simonihmig** [comment]
Thanks for the review and code improvements! I'll rebase and cleanup the next PR now...

**Pull request merged**